src/pymor/models/iosys.py

```diff
@@ -750,70 +765,49 @@ def impulse_resp(self, mu=None, return_solution=False):
 
         Returns
         -------
-        y
+        output
             Impulse response as a 3D |NumPy array| where
-            `y.shape[0]` is the number of time steps,
-            `y.shape[1]` is the number of outputs, and
-            `y.shape[2]` is the number of inputs.
-        Xs
+            `output.shape[0]` is the number of time steps,
+            `output.shape[1]` is the number of outputs, and
+            `output.shape[2]` is the number of inputs.
+        solution
             The tuple of solution |VectorArrays| for every input.
             Returned only when `return_solution` is `True`.
         """
         assert self.T is not None
 
-        if not isinstance(mu, Mu):
-            mu = self.parameters.parse(mu)
+        n = (self.num_values if self.num_values else self.time_stepper.estimate_time_step_count(0, self.T)) + 1
```

Comment:
This should probably be computed the same as in `_compute`.

Reply:
It should be the same (notice the parentheses). Or do you mean the `NotImplementedError`?

Reply:
> Or do you mean the `NotImplementedError`?

Yes, that is what I meant.